pkg/k8s/k8s.go

```diff
@@ -95,6 +82,17 @@ func NewDeployment(n *v1alpha1.Nginx) (*appv1.Deployment, error) {
 		securityContext.RunAsGroup = rootUID
 	}
 
+	if n.Spec.Replicas == nil {
+		var one int32 = 1
+		n.Spec.Replicas = &one
+	}
+
+	// Round up instead of down as is the default behavior for maxUnvailable,
+	// this is useful because we must allow at least one pod down for
+	// hostNetwork deployments.
+	maxUnavailable := intstr.FromInt(int(math.Ceil(float64(*n.Spec.Replicas) * 0.25)))
```

Comment:
I think it's safer applying this change only when hostNetwork is set. Otherwise, we will assume that an Nginx object w/ only one replica may take a short period of downtime.

Reply:
Sure